### PR TITLE
Demo app cleanup

### PR DESCRIPTION
**Pull request by kevinhinterlong**
## Description
The app would fail sometimes when trying to refresh/share it etc because it couldn't go directly to the url. The data returned was random every time (e.g. the same query), and only basic dimension filters could be applied

## Proposed Changes
- Use `hash` locationType to get around url issues
- Set the seed for `faker` when returning a metric value based on the `[day, dimensions requested, current metric]`, so that we can change `filters`/`havings`/`interval` without changing the data
- Added support for more dimension filters `[in, notin, contains]`
- Added support for havings `[gt, gte, lt, lte, eq, neq, bet, nbet]`

## Screenshots
| 15 days query | 14 days query |
|---------|---------|
| 15 day results        |     14 day results     |

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/data/addon/mirage/routes/bard-lite.js [hunk 1]
```diff
@@ -14,7 +14,22 @@ const API_DATE_FORMAT = 'YYYY-MM-DD HH:mm:ss.SSS',
...
-  ];
+  ],
+  HAVING_OPS = {
+    gt: (values, metricValue) => parseFloat(metricValue) > parseFloat(values[0]),
+    gte: (values, metricValue) => parseFloat(metricValue) >= parseFloat(values[0]),
+    lt: (values, metricValue) => parseFloat(metricValue) < parseFloat(values[0]),
+    lte: (values, metricValue) => parseFloat(metricValue) <= parseFloat(values[0]),
+    eq: (values, metricValue) => parseFloat(metricValue) === parseFloat(values[0]),
+    neq: (values, metricValue) => parseFloat(metricValue) !== parseFloat(values[0]),
+    bet: (values, metricValue) => parseFloat(values[0]) < metricValue && metricValue < parseFloat(values[1]),
```
Comment (jkusa) [inline packages/data/addon/mirage/routes/bard-lite.js]: Need more `parseFloat` :)

File: packages/data/addon/mirage/routes/bard-lite.js [hunk 2]
```diff
@@ -137,10 +167,15 @@ export default function(
         if (currFilter.length > 0) {
           if (currFilter[0] === ',') currFilter = currFilter.substring(1);
 
-          let dimension = currFilter.split('|')[0],
-            values = currFilter.split('[')[1].split(',');
+          const columnOp = currFilter.split('-');
+          const column = columnOp[0].split('|');
+          const op = columnOp[1].split('[');
+          let dimension = column[0],
```
Comment (jkusa) [inline packages/data/addon/mirage/routes/bard-lite.js]: `const` all the things

File: packages/data/addon/mirage/routes/bard-lite.js [hunk 3]
```diff
@@ -156,27 +191,60 @@ export default function(
...
+    let havings = {};
+    if (request.queryParams.having) {
+      havings = request.queryParams.having.split(']').reduce((havingObj, currHaving) => {
+        if (currHaving.length > 0) {
+          if (currHaving[0] === ',') currHaving = currHaving.substring(1);
+
+          const rowOp = currHaving.split('-');
+          const op = rowOp[1].split('[');
+          const metric = rowOp[0],
+            operation = op[0],
```
Comment (jkusa) [inline packages/data/addon/mirage/routes/bard-lite.js]: `const`

File: packages/data/addon/mirage/routes/bard-lite.js [hunk 4]
```diff
@@ -156,27 +191,60 @@ export default function(
...
+      }, []);
     });
 
+    let havings = {};
+    if (request.queryParams.having) {
+      havings = request.queryParams.having.split(']').reduce((havingObj, currHaving) => {
+        if (currHaving.length > 0) {
+          if (currHaving[0] === ',') currHaving = currHaving.substring(1);
+
+          const rowOp = currHaving.split('-');
```
Comment (jkusa) [inline packages/data/addon/mirage/routes/bard-lite.js]: can we do regex extract?

File: packages/data/addon/mirage/routes/bard-lite.js (hunk 2 above)
Comment (kevinhinterlong) [reply packages/data/addon/mirage/routes/bard-lite.js]: giving up on `const` in order to use regex extraction and make small fixups to `field`and `values`

**Pull request merged**